### DISCUSSION TITLE
Segmentation Fault when creating ARRAY logical type in C API

### DISCUSSION THREAD

**Discussion by luke-griffiths** [duckdb/duckdb#15185] [Q&A] [answered]
Hi! I'm using the C API to append rows to a table with a DOUBLE[3] column. I'm getting a segmentation fault when I try to create the `duckdb_logical_type` for that column. Example code is below.
```
duckdb_appender appender;
// next line seg faults
duckdb_logical_type arr_type = duckdb_create_array_type(DUCKDB_TYPE_DOUBLE, 3);

duckdb_data_chunk chunk = duckdb_create_data_chunk(arr_type, 1);
// .. code to change data in chunk and append
// ..
duckdb_destroy_logical_type(&arr_type);
duckdb_destroy_data_chunk(&chunk);
```
It isn't obvious to me why calling `duckdb_create_array_type` would cause a segfault. Is there an alternative way to append arrays using the C API?

Using the most recent release of C API on x86

**Comment by Tishj** [comment]
Discussions are not the place to raise bugs, it will get overlooked

You're saying that this is all that's required to segfault?
`duckdb_logical_type arr_type = duckdb_create_array_type(DUCKDB_TYPE_DOUBLE, 3);`

**Reply by Tishj** [reply, marked as answer]
Ah nvm this is user error, you're giving a `duckdb_type` as the first argument, not a `duckdb_logical_type`

**Reply by luke-griffiths** [reply]
Thank you for pointing that out!
